From 5fd96ac5c7192cdf36bc5b543dfc9e75503bba2a Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Wed, 20 Mar 2024 19:34:00 +0100
Subject: [PATCH 1030/1141] media: dcmipp: addition of BGR888 format output
 from pixel pipes

BGR888 and RGB888 outputs are same in case of PPCR format configuration
and only swap R / B should be adjusted to switch from BGR888 to RGB888.

Change-Id: Iaa95e9fe8def5fe106a61cac03339378bac2ddf1
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/368125
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelcap.c  | 2 ++
 drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c | 1 +
 2 files changed, 3 insertions(+)

--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelcap.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelcap.c
@@ -107,6 +107,7 @@ static const struct dcmipp_pixelcap_pix_
 	PIXMAP_MBUS_PFMT(VYUY8_2X8, VYUY, 1),
 	PIXMAP_MBUS_PFMT(Y8_1X8, GREY, 1),
 	PIXMAP_MBUS_PFMT(RGB888_1X24, RGB24, 1),
+	PIXMAP_MBUS_PFMT(BGR888_1X24, BGR24, 1),
 	PIXMAP_MBUS_PFMT(ARGB8888_1X32, ARGB32, 1),
 	PIXMAP_MBUS_PFMT(AYUV8_1X32, AYUV32, 1),
 
@@ -236,6 +237,7 @@ static inline int hdw_pixel_alignment(u3
 	case V4L2_PIX_FMT_VYUY:
 		return 3;/* 2^3  = 8 pixels = 16 bytes */
 	case V4L2_PIX_FMT_RGB24:
+	case V4L2_PIX_FMT_BGR24:
 		return 4;/* 2^4 = 16 pixels = 48 bytes */
 	case V4L2_PIX_FMT_ARGB32:
 		return 2;/* 2^2  = 4 pixels = 16 bytes */
--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
@@ -147,6 +147,7 @@ static const struct dcmipp_pixelproc_pix
 
 static const struct dcmipp_pixelproc_pix_map dcmipp_pixelproc_src_pix_map_list[] = {
 	PIXMAP_MBUS_PPCR_SWAPUV(RGB888_1X24, DCMIPP_PxPPCR_FORMAT_RGB888_OR_YUV444_1BUFFER, 1),
+	PIXMAP_MBUS_PPCR_SWAPUV(BGR888_1X24, DCMIPP_PxPPCR_FORMAT_RGB888_OR_YUV444_1BUFFER, 0),
 	PIXMAP_MBUS_PPCR_SWAPUV(RGB565_2X8_LE, DCMIPP_PxPPCR_FORMAT_RGB565, 0),
 	PIXMAP_MBUS_PPCR_SWAPUV(YUYV8_2X8, DCMIPP_PxPPCR_FORMAT_YUYV, 0),
 	PIXMAP_MBUS_PPCR_SWAPUV(YVYU8_2X8, DCMIPP_PxPPCR_FORMAT_YUYV, 1),
